frontend/src/components/App/Home/index.tsx

```diff
@@ -254,103 +76,39 @@ function HomeComponent(props: HomeComponentProps) {
 
   React.useEffect(() => {
     setCustomNameClusters(currentNames => {
-      if (isEqual(currentNames, getClusterNames())) {
+      if (isEqual(currentNames, getCustomClusterNames(clusters))) {
         return currentNames;
       }
-      return getClusterNames();
+      return getCustomClusterNames(clusters);
     });
   }, [customNameClusters]);
 
-  function getClusterNames() {
-    return Object.values(clusters)
-      .map(c => ({
-        ...c,
-        name: c.meta_data?.extensions?.headlamp_info?.customName || c.name,
-      }))
-      .sort();
-  }
-
-  /**
-   * Gets the origin of a cluster.
-   *
-   * @param cluster
-   * @returns A description of where the cluster is picked up from: dynamic, in-cluster, or from a kubeconfig file.
-   */
-  function getOrigin(cluster: Cluster): string {
-    if (cluster.meta_data?.source === 'kubeconfig') {
-      const kubeconfigPath = process.env.KUBECONFIG ?? '~/.kube/config';
-      return `Kubeconfig: ${kubeconfigPath}`;
-    } else if (cluster.meta_data?.source === 'dynamic_cluster') {
-      return t('translation|Plugin');
-    } else if (cluster.meta_data?.source === 'in_cluster') {
-      return t('translation|In-cluster');
-    }
-    return 'Unknown';
-  }
-
   const memoizedComponent = React.useMemo(
     () => (
       <PageGrid>
-        <SectionBox headerProps={{ headerStyle: 'main' }} title={t('Home')}>
-          <RecentClusters clusters={Object.values(customNameClusters)} onButtonClick={() => {}} />
-        </SectionBox>
+        {ENABLE_RECENT_CLUSTERS && (
+          <SectionBox headerProps={{ headerStyle: 'main' }} title={t('Home')}>
+            <RecentClusters clusters={Object.values(customNameClusters)} onButtonClick={() => {}} />
+          </SectionBox>
+        )}
         <SectionBox
           title={
             <SectionFilterHeader
               title={t('All Clusters')}
               noNamespaceFilter
-              headerStyle="subsection"
+              {...(ENABLE_RECENT_CLUSTERS
```

Comment:
`headerStyle={ENABLED_RECENT_CLUSTERS ? 'subsection' : 'main'}` 
I think this version would be a bit more readable

Reply:
done